### PR TITLE
Word count template in Go

### PR DESCRIPTION
**Pull request by diego-aquino**
This pull request implements a template in Go of the word count program based on [wc_single_proc.c](https://github.com/thiagomanel/SO/blob/master/process/word_count/wc_single_proc.c). The script [make_dataset.sh](https://github.com/thiagomanel/SO/blob/master/process/word_count/make_dataset) and the file [file2count.data](https://github.com/thiagomanel/SO/blob/master/process/word_count/file2count.data) are also included.

@thiagomanel

### CODE REVIEW

File: 2023.1/word_count/go/word_count.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"unicode"
+)
+
+// Count the number of words in `fileContent`.
+func wc(fileContent string) int {
```
Comment (thiagomanel) [inline 2023.1/word_count/go/word_count.go:10]: maybe refactor this function to use `SplitAfter`?

**Review by thiagomanel — CHANGES_REQUESTED** [review]
thanks. I think we can improve a little bit the `wc` function

**Review by thiagomanel — APPROVED** [review]
thanks

**Comment by thiagomanel** [comment]
> thanks

ignore. A send the message in the wrong PR. Still waiting for the refactored ode :)

File: 2023.1/word_count/go/word_count.go (hunk 1 above)
Comment (diego-aquino) [reply 2023.1/word_count/go/word_count.go:10]: Alright, I'll take a look!

Comment (diego-aquino) [reply 2023.1/word_count/go/word_count.go:10]: @thiagomanel, I've found this native function that splits a string by its whitespaces: https://pkg.go.dev/strings#Fields. What do you think?

File: 2023.1/word_count/go/word_count.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+// │     ├── file
+// │     ├── ...
+// │     └── file
+// ├── ...
+// └── subdir N
+// │     ├── file
+// │     ├── ...
+// │     └── file
+func main() {
+	// Implement here...
```
Comment (diego-aquino) [inline 2023.1/word_count/go/word_count.go:68]: @thiagomanel, to make sure, the main should be empty, right?

File: 2023.1/word_count/go/word_count.go (hunk 1 above)
Comment (thiagomanel) [reply 2023.1/word_count/go/word_count.go:10]: it seams a good option. go ahead

File: 2023.1/word_count/go/word_count.go (hunk 2 above)
Comment (thiagomanel) [reply 2023.1/word_count/go/word_count.go:68]: maybe we can code the main function like the old example: it receives a root dir and call wc_dir for each subdirectory.

Comment (diego-aquino) [reply 2023.1/word_count/go/word_count.go]: Okay, I just added the implementation of `main` based on the example. Could you take a look?

**Review by thiagomanel — APPROVED** [review]
lgtm

**Pull request merged**